configmap/informer/informed_watcher.go

```diff
@@ -207,3 +241,91 @@ func (i *InformedWatcher) deleteConfigMapEvent(obj interface{}) {
 	}
 	// If there is no default value, then don't do anything.
 }
+
```

Comment:
* I'm not sure if there is a better package to put these kind of synchronization primitives in?
* Instead of making these wrappers around a WaitGroup, I could use polling (e.g. like `cache.WaitForCacheSync(stopCh, i.informer.Informer().HasSynced)`) but using go synchronization primitives to do synchronization between goroutines seemed like a better solution to me (less consistent with the surrounding code tho)

Reply:
writing tests for unexported code gives me a no feeling but i can understand why you dont want ppl to start importing this and relying on it :D